pkg/shares/share_merging.go

```diff
@@ -141,6 +142,25 @@ func ParseShares(rawShares [][]byte) ([]ShareSequence, error) {
 	return sequences, nil
 }
 
+// RawData returns the raw share data of this share sequence. The raw data does
+// not contain the namespace ID, info byte, sequence length, or reserved bytes.
+func (s ShareSequence) RawData() (data []byte, err error) {
+	for _, share := range s.Shares {
+		raw, err := share.RawData()
+		if err != nil {
+			return []byte{}, err
+		}
+		data = append(data, raw...)
```

Comment:
[forreals-don't-change-if-you-don't-feel-like-it-I-really-dgaf-this-code-is-nice-and-readable]
the optimization linter fails again!

Reply:
oh true, https://github.com/celestiaorg/celestia-app/blob/b65353f456335ed195c67522d4ec546dc3a583fb/.golangci.yml#L11 is enabled so 🤷‍♂️